components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -949,6 +949,8 @@ impl CanvasRenderingContext2DMethods for CanvasRenderingContext2D {
                 self.ipc_renderer.send(msg).unwrap();
             },
             StringOrCanvasGradientOrCanvasPattern::eCanvasPattern(pattern) => {
+                self.state.borrow_mut().stroke_style =
+                    CanvasFillOrStrokeStyle::Pattern(JS::from_ref(pattern.r()));
```

Comment:
I am not sure the benefit of using `JS::from_ref(pattern.r())` vs `JS::from_rooted(pattern)`, I have just kept consistent with the other matches